planner/core/task.go

```diff
@@ -2228,21 +2242,37 @@ func (t *mppTask) enforceExchangerImpl(prop *property.PhysicalProperty) *mppTask
 		for _, col := range prop.MPPPartitionCols {
 			if types.IsString(col.Col.RetType.GetType()) {
 				t.p.SCtx().GetSessionVars().RaiseWarningWhenMPPEnforced("MPP mode may be blocked because when `new_collation_enabled` is true, HashJoin or HashAgg with string key is not supported now.")
-				return &mppTask{}
+				return &mppTask{mppVersion: t.mppVersion}
 			}
 		}
 	}
+
 	ctx := t.p.SCtx()
 	sender := PhysicalExchangeSender{
 		ExchangeType: prop.MPPPartitionTp.ToExchangeType(),
 		HashCols:     prop.MPPPartitionCols,
+		MppVersion:   t.mppVersion,
 	}.Init(ctx, t.p.statsInfo())
+
+	if sender.MppVersion >= kv.MppVersionV1 {
+		// Use compress when exchange tyoe is `Hash`
+		if sender.ExchangeType == tipb.ExchangeType_Hash {
+			mode := ctx.GetSessionVars().MppExchangeCompressionMode
```

Comment:
When in explain/explain analyze statement, add a warning when user set compression mode to ExchangeCompressionModeFast or ExchangeCompressionModeHC but current mpp version does not support compression?

Reply:
OK. I've add warning when setting `mpp_exchange_compression_mode`.